tools/timemachine/internal/cmd/downloadHeight.go

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"google.golang.org/api/iterator"
+)
+
+// const value
+var (
+	GcpTimeout  = time.Second * 60
+	ErrNotExist = errors.New("height is not existed.")
+)
+
+// downloadHeight represents the download command
+var downloadHeight = &cobra.Command{
+	Use:   "download",
+	Short: "Download specific height of block",
+	Args:  cobra.ExactArgs(2),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		height, err := strconv.ParseUint(args[0], 10, 64)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to convert input height: %s.", args[0])
+		}
+		if height == 0 {
+			return errors.New("input height cannot be 0.")
+		}
+		var (
+			heighM    = float64(height) / float64(1024) / float64(1024)
+			heightDir string
+			destDir   = args[1]
+			bucket    = "blockchain-golden"
+			prefix    = "fullsync/mainnet/"
+			wg        sync.WaitGroup
+		)
+
+		allObjNames, err := listFiles(bucket, "", "")
+		if err != nil {
+			return err
+		}
+		if len(allObjNames) == 0 {
+			return ErrNotExist
+		}
+		heiNames := make(map[string][]string)
+		curr := 0.0
+		for _, name := range allObjNames {
+			if strings.HasPrefix(name, prefix) {
+				s := strings.TrimPrefix(name, prefix)
+				if len(s) > 0 {
+					heiStr := s[:strings.Index(s, "/")]
+					_, ok := heiNames[heiStr]
+					if ok {
+						heiNames[heiStr] = append(heiNames[heiStr], name)
+					} else {
+						heiNames[heiStr] = make([]string, 0)
+					}
+
+					if heiStr != "latest" {
+						v := convertHeightStr(heiStr)
+						if v > curr {
+							curr = v
+						}
+					}
+				}
+			}
+		}
+		latest := curr + 0.25
+
+		if heighM < 8 {
```

Comment:
no need to divide by 100000, just use the height
0-8000000
8000000-12000000
12000000-13000000
13000000-13250000
etc.

Reply:
fixed